src/Context/ContextDefinitionInterface.php

```diff
@@ -10,8 +10,33 @@
 use \Drupal\Core\Plugin\Context\ContextDefinitionInterface as ContextDefinitionInterfaceCore;
 
 /**
- * Extends the core context definition interface.
+ * Context definition information required by Rules.
+ *
+ * The core interface is extended to add properties that are necessary for
+ * Rules.
  */
 interface ContextDefinitionInterface extends ContextDefinitionInterfaceCore {
 
+  /**
+   * Determines if the context value is allowed to be NULL.
+   *
+   * Usually Rules will not pass any NULL values as argument, but abort the
```

Comment:
Let's better keep docs general here as the interface is general and does not enforce usage by Rules also. We might be able to move improvements in core later etc., docs should be written to facilitate that imho.